luigi/scheduler.py

```diff
@@ -556,6 +556,12 @@ def get_worker_ids(self):
     def get_worker(self, worker_id):
         return self._active_workers.setdefault(worker_id, Worker(worker_id))
 
+    def get_worker_starts(self):
```

Comment:
Turns out this function really slows down the quadratic behavior test because it makes count_pending linear in the number of workers. I've instead written a quick SortedSet data structure so we can just check the last worker id in Task.workers.